### PR TITLE
Update COM host to match RegAsm registration behavior

### LINKED ISSUE #62529: How to change the ComHost Server registry name?
I'm working on a plugin-based application architecture that relies on custom naming of the COM server classes in the registry. The `<project>.comhost.dll` COM server generated for projects relying on `<EnableComHosting>` today always seem to be named `CoreCLR COMHost Server` which says _nothing_ about the COM server. I am therefore seeking a mechanism for either more informative default naming or an ability to override the name.

Is there a mechanism for customizing the COMHost server name to something else than "CoreCLR COMHost Server"?

### Steps to reproduce
1. Build the https://github.com/dotnet/samples/tree/main/core/extensions/COMServerDemo/COMServer
2. Register `COMServer.comhost.dll` with regsvr32.exe
3. Open RegEdit.exe and observe that the server name is "CoreCLR COMHost Server" in the Computer\HKEY_CLASSES_ROOT\CLSID\{DB1797F5-7198-4411-8563-D05F4E904956} registry folder:
image

### .Net framework behavior
The .Net framework 4.x solved this problem for projects enabling "Register for COM interop" by **using the same ProgId string also for the COM server name**. Using the same naming scheme also in .Net core would be an acceptable outcome for me.

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Fixes #62529

/cc @jkoritzinsky @elinor-fung

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
We should consider this fix for servicing in .NET 6.

File: src/native/corehost/comhost/comhost.cpp [hunk 1]
```diff
@@ -431,14 +430,14 @@ namespace
 
             regKey.reset(regProgIdKeyRaw);
 
-            // The default value for the key is the ProgID
+            // The default value for the key is the type - this matches RegAsm behavior.
```
Comment (elinor-fung) [inline src/native/corehost/comhost/comhost.cpp:433]: We ignore what the user has specified as the ProgID and use the type name to set `SOFTWARE\Classes\CLSID\<clsid>\ProgID`, but add a key under `SOFTWARE\Classes\` using the specified ProgID? Seems odd to me - is this seriously what RegAsm does?

Comment (AaronRobinsonMSFT) [reply src/native/corehost/comhost/comhost.cpp:433]: ...

Happy new year!

**Review by elinor-fung — APPROVED** [review]

**Pull request merged**